ereuse_devicehub/inventory/forms.py

```diff
@@ -0,0 +1,87 @@
+from flask_wtf import FlaskForm
+from wtforms import StringField, validators
+from flask import g
+
+from ereuse_devicehub.db import db
+from ereuse_devicehub.resources.device.models import Device
+from ereuse_devicehub.resources.lot.models import Lot
+
+
+class LotDeviceForm(FlaskForm):
+    lot = StringField(u'Lot', [validators.UUID()])
+    devices = StringField(u'Devices', [validators.length(min=1)])
+
+    def validate(self, extra_validators=None):
+        is_valid = super().validate(extra_validators)
+
+        if not is_valid:
+            return False
+
+        self._lot = Lot.query.filter(Lot.id == self.lot.data).filter(
+            Lot.owner_id == g.user.id).one()
+
+        devices = set(self.devices.data.split(","))
+        self._devices = set(Device.query.filter(Device.id.in_(devices)).filter(
+            Device.owner_id == g.user.id).all())
```

Comment:
Use SQL distinct[1] instead of set()
https://docs.sqlalchemy.org/en/12/orm/query.html#sqlalchemy.orm.query.Query.distinct